skglm/experimental/reweighted.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+from numpy.linalg import norm
+from skglm.estimators import Lasso
+
+
+def _reweight_by_coefficient_norm(coef):
```

Comment:
What about calling it `L05_weights` ?
this way, we'd pave the way for a more fexible design where a penalty is passed, and penalty.weights() is called. 
WDYT?

this would allows us to solve MCP or L23 with reweighting with a simple change

Reply:
Agreed!